service/controller/kvm/v7/resource/chartconfig/current.go

```diff
@@ -29,13 +30,23 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		ClusterID: key.ClusterID(clusterGuestConfig),
 	}
 	chartConfigs, err := r.chartConfig.GetCurrentState(ctx, clusterConfig)
-	if guest.IsAPINotAvailable(err) {
+	if apierrors.IsNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the chartconfig CRD in the tenant cluster")
+
+		// ChartConfig CRD is created by chart-operator which may not be
+		// running yet in the tenant cluster. We will retry during the next
+		// execution.
+		resourcecanceledcontext.SetCanceled(ctx)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+		return nil, nil
+	} else if guest.IsAPINotAvailable(err) {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "tenant cluster is not available")
 
 		// We can't continue without a successful K8s connection. Cluster
 		// may not be up yet. We will retry during the next execution.
-		reconciliationcanceledcontext.SetCanceled(ctx)
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
+		resourcecanceledcontext.SetCanceled(ctx)
```

Comment:
So why does the cancelation change now? I don't follow here.

Reply:
My thinking was that the chartconfig resource is the final resource. So there is nothing after to stop. But I guess we could another resource later.

I reverted as its an unnecessary change. Also if we can't connect to the tenant there is nothing useful that can be done.